# Introduction

### (What is ionomics) 

	*The ionome provides a snapshot of the functional status of a complex biological organism. Ionomics can precisely capture information about the functional state of an organism under different conditions, which may be driven by differences in genetics, developmental stage, or exposure to distinct biotic or abiotic factors.
	* As sessile organisms, plants have evolved sophisticated mechanisms to adapta and survive in a diverse range of environments, including hihgly heterogenous soils. Plants take up most of their mineral nutrients and trace elements they require from the soil. Thus, the ionomic profiles of plants grown in a common environment may reflect genetic adaptations to their native local environments (Baxter et al 2010; Anderson & Mitchell-Olds 2010). (Huang & Salt 2016) 
	
### (GxE definition) 

	* Local adaptation is the process by which natural selection drives adaptive phenotypic divergence across environmental gradients. 
	* Local adaptation could be driven, at least in some cases, by ionomic loci.
	* The ionomic snapshot depends not only on the genetic makeup of the individual, but also on the environmental context that the individual experiences.
	* Genetic variation in the makeup of these ionomic snapshots between environments is a type of gene by environment interactions (GxE).
	* Evolutionary theory suggests that local adaptation is driven by genetic tradeoffs in phenotypic expression between environments. 
	* Two extremes commonly invoked for types of GxE are antagonistic pleiotropy (AP) and conditional neutrality. An allele with AP increases phenotypic expression in one environment and decreases it in a second environment, relative to an allele without GxE. The change in sign of this allelic effect could occur because of costs of phenotypic expression that vary by environment, or tradeoffs to phenotypic expression. An allele with conditional neutrality alters the phenotype in one environment without affecting it in a second environment, relative to an allele without GxE. Conditional neutrality is the most extreme case of differential sensitivity (DS), which occurs when an allele increases, or decreases, phenotypic expression in one environment more than in a second environment, relative to an allele without GxE. When DS alleles have been identified, transcription factors and transporters are often found, as well as genes with allelic variation in regulatory sequences that causes induced gene expression (Mickelbart et al 2015). 
	* It seems plausible that ionomic phenotypes could also be regulated or induced by particular phenotypic conditions.

### (Studying the molecular mechanisms of GxE has been difficult) 

	* many previous studies focused on a limited environmental range
	* or were restricted to biparental crosses in annual, selfing species
	* or relied on only two field sites, often at the extreme ends of environmental gradients
	* how do the molecular responses to the environment change across environmental gradients?
	* Understanding these responses could help us link the genotypic and phenotypic maps (Hoekstra cite).

### (How we circumvent these difficulties: *Panicum virgatum* four-way cross) 

	* have ten field sites that cover 17 degrees of latitude in the central United States
	* use outbreeding perennial species, which can be cloned so identical genotypes can be planted at the ten sites.
	* have a cross involving four parents: three switchgrass cultivars and one wild accession. These parents are from two highly divergent southern lowland and northern upland ecotypes. (add info on ecotypes).
	* planted 425 clones from a four-way outbred mapping population at all ten field sites. Also planted clones of all four grandparents and F1 hybrid plants of the mapping population.	

### (Here, we do x, y, and z) 

	* Here, we are able to address the following major questions:
	i) How does variation in environmental conditions across geographic space influence ionomic trait variation?
	ii) How often are QTL affecting ionomic traits subject to genotype x environment (GxE) interactions?
	iii) How many QTL correspond to known candidate genes, and how many to novel genomic regions?
